### PR TITLE
Clear command

### PR DESCRIPTION
**Pull request by 0xfedcafe**
Added clear command and test as suggested in issue #1810

### CODE REVIEW

File: vtkext/private/module/vtkF3DImguiConsole.h [hunk 1]
```diff
@@ -43,6 +43,16 @@ class vtkF3DImguiConsole : public vtkF3DConsoleOutputWindow
...
 
+  /**
+   * Clear console
+   */
+  void Clear();
+
+  /**
+   * Get console history size
+   */
+  size_t GetLogsSize();
```
Comment (mwestphal) [inline vtkext/private/module/vtkF3DImguiConsole.h:54]: why is this method needed ?

File: library/testing/TestSDKInteractorCommand.cxx [hunk 2]
```diff
@@ -24,6 +24,12 @@ int TestSDKInteractorCommand(int argc, char* argv[])
   inter.triggerCommand("reset render.hdri.file");
   test("triggerCommand reset optional", options.render.hdri.file.has_value() == false);
 
+  // Test clear
+#if F3D_MODULE_UI
+  inter.triggerCommand("print_scene_info");
+  test("triggerCommand clear", (inter.triggerCommand("clear") == true));
+#endif
```
Comment (mwestphal) [inline library/testing/TestSDKInteractorCommand.cxx:31]: While this is an ok way to test this, a baseline based test (in application/testing/CMakeLists.txt) would be better imo.

**Review by mwestphal — CHANGES_REQUESTED** [review]
thanks for the contribution! Some changes needed though :)

Also, you need to update the doc in `doc/user/COMMANDS.md`

File: vtkext/private/module/vtkF3DImguiConsole.h (hunk 1 above)
Comment (0xfedcafe) [reply vtkext/private/module/vtkF3DImguiConsole.h:54]: it's not needed, i forgot to remove it after trying to write a test that somehow directly accesses the vtkF3DImguiConsole, sorry 😅

File: application/testing/CMakeLists.txt [hunk 3]
```diff
@@ -821,6 +821,7 @@ f3d_test(NAME TestInteractionConsoleOpenWarningClick DATA f3d.glb ARGS --positio
 f3d_test(NAME TestInteractionConsoleToggleGrid DATA f3d.glb INTERACTION UI) #Escape;toggle render.grid.enable;Return;Escape
 f3d_test(NAME TestInteractionConsoleInvalidCommand DATA f3d.glb INTERACTION UI) #Escape;foo;Return
 f3d_test(NAME TestInteractionConsoleInvalidOption DATA f3d.glb INTERACTION UI) #Escape;toggle foo;Return
+f3d_test(NAME TestInteractionConsoleClear DATA dragon.vtu INTERACTION UI) #Escape;clear;Return
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:824]: lets write another command first to make sure it is cleared by the `clear` command ?

File: library/src/interactor_impl.cxx [hunk 4]
```diff
@@ -603,6 +608,19 @@ interactor& interactor_impl::initCommands()
...
       this->Internals->Options.reset(args[0]);
     });
+  this->addCommand("clear",
+    [&](const std::vector<std::string>& args)
+    {
+      check_args(args, 0, "clear");
+#if F3D_MODULE_UI
+      vtkF3DImguiConsole* console =
+        vtkF3DImguiConsole::SafeDownCast(vtkOutputWindow::GetInstance());
+      if (console)
```
Comment (mwestphal) [inline library/src/interactor_impl.cxx:618]: I think an assert is enough here ?

File: testing/recordings/TestInteractionConsoleClear.log [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+
+KeyPressEvent 0 0 0 114 1 r 0
+CharEvent 0 0 0 114 1 r 0
+KeyReleaseEvent 0 0 0 114 1 r 0
+
+
+KeyPressEvent 0 0 0 13 1 Return 0
+CharEvent 0 0 0 13 1 Return 0
+KeyReleaseEvent 0 0 0 13 1 Return 0
```
Comment (mwestphal) [inline testing/recordings/TestInteractionConsoleClear.log:129]: missing end of line

File: vtkext/private/module/vtkF3DImguiActor.h [hunk 6]
```diff
@@ -86,6 +86,11 @@ class vtkF3DImguiActor : public vtkF3DUIActor
    */
   void RenderConsoleBadge() override;
 
+  /**
+   * Clear console
+   */
+  void Clear() override;
```
Comment (mwestphal) [inline vtkext/private/module/vtkF3DImguiActor.h:92]: this method is not needed either.

File: application/testing/CMakeLists.txt (hunk 3 above)
Comment (0xfedcafe) [reply application/testing/CMakeLists.txt:824]: Oops, I forgot to change the comment, it actually executes print_scene_info before calling clear as you can see in log file, if I understand correctly how it works.

File: vtkext/private/module/vtkF3DUIActor.h [hunk 7]
```diff
@@ -96,6 +96,11 @@ class vtkF3DUIActor : public vtkProp
    */
   virtual void SetDeltaTime(double) {}
 
+  /**
+   * Clear console
+   */
+  virtual void Clear() {}
```
Comment (mwestphal) [inline vtkext/private/module/vtkF3DUIActor.h:102]: you can remove this

File: application/testing/CMakeLists.txt [hunk 8]
```diff
@@ -821,6 +821,7 @@ f3d_test(NAME TestInteractionConsoleOpenWarningClick DATA f3d.glb ARGS --positio
 f3d_test(NAME TestInteractionConsoleToggleGrid DATA f3d.glb INTERACTION UI) #Escape;toggle render.grid.enable;Return;Escape
 f3d_test(NAME TestInteractionConsoleInvalidCommand DATA f3d.glb INTERACTION UI) #Escape;foo;Return
 f3d_test(NAME TestInteractionConsoleInvalidOption DATA f3d.glb INTERACTION UI) #Escape;toggle foo;Return
+f3d_test(NAME TestInteractionConsoleClear DATA dragon.vtu INTERACTION UI) #Escape;e;Escape;printt;BackSpace;_scee;Left;Right;_info;Return;clear;Return
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:824]: The interaction sequence is a bit overkill but thats alright :)

```suggestion
f3d_test(NAME TestInteractionConsoleClear DATA dragon.vtu INTERACTION UI) #Escape;e;Escape;printt;BackSpace;_scee;Left;n;Right;_info;Return;clear;Return
```

File: application/testing/CMakeLists.txt [hunk 9]
```diff
@@ -821,6 +821,7 @@ f3d_test(NAME TestInteractionConsoleOpenWarningClick DATA f3d.glb ARGS --positio
...
 f3d_test(NAME TestInteractionConsoleTypingSceneInfo DATA f3d.glb INTERACTION UI) #Escape;e;Escape;printt;BackSpace;_scee;Left;Right;_info;Return
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:825]:
```suggestion
f3d_test(NAME TestInteractionConsoleTypingSceneInfo DATA f3d.glb INTERACTION UI) #Escape;e;Escape;printt;BackSpace;_scee;Left;n;Right;_info;Return
```

**Review by Meakk — APPROVED** [review]
It looks great! Thanks

**Review by mwestphal — APPROVED** [review]

**Comment by mwestphal** [comment]
Thanks for your contribution @0xfedcafe !!!

**Pull request merged**